File: solutions/ocaml/darts/1/darts.ml
let score (x: float) (y: float): int =
  let r = sqrt (x *. x +. y *. y) in
  match r with
  | r when r <= 1.0 -> 10
  | r when r <= 5.0 -> 5
  | r when r <= 10.0 -> 1
  | _ -> 0

File: solutions/clojure/darts/1/src/darts.clj
(ns darts)

(defn score
  "Calculates the score of a dart throw."
  [x y]
  (let [rad (Math/sqrt (+ (* x x) (* y y)))]
    (cond
      (<= rad 1) 10
      (<= rad 5) 5
      (<= rad 10) 1
      :else 0)))
